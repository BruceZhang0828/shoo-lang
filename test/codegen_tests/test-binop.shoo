int x = 5+6;
println(str_of_int(x));
int y = 2*3;
println(str_of_int(y));

int z = y % 4;
println(str_of_int(z));

z = z % 4;
println(str_of_int(z));

bool q = z < x;
println(str_of_bool(z < x));

q = z > x; 
println(str_of_bool(q));

q = z >= 2;
println(str_of_bool(q));

q = z <= 2;
println(str_of_bool(q));

if (x==y) {
	println("SHOULDNT PRINT THIS");
} else {
	println("yes good");
}

if (x!=y) {
	println("yes good");
} else {
	println("SHOULDNT PRINT THIS");
}
