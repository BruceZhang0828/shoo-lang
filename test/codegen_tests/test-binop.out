11
6
2
2
true
false
true 
true
yes good
yes good
